numba/typed/dictobject.py

```diff
@@ -242,7 +247,7 @@ def codegen(context, builder, sig, args):
         refdp = cgutils.alloca_once(builder, ll_dict_type, zfill=True)
         status = builder.call(
             fn,
-            [refdp, ll_ssize_t(sz_key), ll_ssize_t(sz_val)],
+            [refdp, args[2], ll_ssize_t(sz_key), ll_ssize_t(sz_val)],
```

Comment:
Does `args[2]` need a cast to a `size_t`?

Reply:
`numba_dict_new_sized` takes a `Py_ssize_t` for `n_keys`:

```C
int numba_dict_new_sized(NB_Dict **out, Py_ssize_t n_keys, Py_ssize_t key_size, Py_ssize_t val_size);
```

so I've changed the code to cast to that. We were getting away with it at least in CI because `args[2]` was an `int64`. Whilst making this change I also edited the `codegen` function so it's more of a straight-line read rather than multi-line-things-happening-in-the-indents code.

Reply:
I want to point out that `listobject.py` also passes the size parameter without casting with `ll_ssize_t()` (as far as I can tell). 

`_list_new` passes `args[1]` to `_list_new_codegen` as `new_size`. There is no casting of this variable. It is then passed to `numba_list_new` in `listobject.c` as the argument `allocated`, which is a `Py_ssize_t`. I imitated this approach. Perhaps this needs to be fixed as well?

Sorry for the lack of responses lately. I've been busy.

Reply:
No problem, and thanks for all your efforts so far! If `_list_new_codegen` does this, then it probably ought to be fixed too. However, as this PR is nearly over the line I'd suggest we do that in a separate PR (if we're going to... As nobody noticed now it seems like we're getting away with it 🙂 )